aws/terraform/infra/vpc.tf

```diff
@@ -0,0 +1,152 @@
+resource "aws_vpc" "this" {
+  cidr_block           = "10.0.0.0/16"
+  enable_dns_hostnames = true
+
+  tags = merge(
+    module.common_vars.tags,
+    {
+      Name = "${local.resource_prefix}-vpc-${local.resource_suffix}"
+    }
+  )
+}
+
+/*
+ Setup a gateway between aws VPC and internet. Allow access to and from Resources
+ in subnet with public IP addr.
+ Ref: https://nickcharlton.net/posts/terraform-aws-vpc.html
+*/
+resource "aws_internet_gateway" "this" {
+  vpc_id = aws_vpc.this.id
+
+  tags = merge(
+    module.common_vars.tags,
+    {
+      Name = "${local.resource_prefix}-internet-gateway-${local.resource_suffix}"
+    }
+  )
+}
+
+/*
+ Create a public Subnet that the two follow private Subnets can use for internet egress connections.
+ Contains our NAT Gateway.
+*/
+resource "aws_subnet" "public" {
+  vpc_id                  = aws_vpc.this.id
+  cidr_block              = "10.0.32.0/20"
+  availability_zone       = "${var.aws_region}a"
```

Comment:
nit: it may be nicer to use  [`aws_availability_zone` data provider](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/availability_zones) instead of constructing the name using string concatenation.

Reply:
Agreed, I'll make this improvement.

Reply:
Agreed, I'll make this improvement.

Reply:
Resolved in my latest commit!